helm/resource_release_test.go

```diff
@@ -56,6 +55,51 @@ func TestAccResourceRelease_basic(t *testing.T) {
 	})
 }
 
+func TestAccResourceRelease_import(t *testing.T) {
+	name := fmt.Sprintf("test-import-%s", acctest.RandString(10))
+	namespace := fmt.Sprintf("%s-%s", testNamespace, acctest.RandString(10))
+	// Delete namespace automatically created by helm after checks
+	defer deleteNamespace(t, namespace)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t, namespace) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckHelmReleaseDestroy(namespace),
+		Steps: []resource.TestStep{{
+			Config: testAccHelmReleaseConfigBasic(testResourceName, namespace, name, "7.1.0"),
+			Check: resource.ComposeAggregateTestCheckFunc(
+				resource.TestCheckResourceAttr("helm_release.test", "metadata.0.revision", "1"),
+				resource.TestCheckResourceAttr("helm_release.test", "metadata.0.version", "7.1.0"),
+				resource.TestCheckResourceAttr("helm_release.test", "status", release.StatusDeployed.String()),
+			),
+		}, {
+			Config:                  testAccHelmReleaseConfigBasic("imported", namespace, "import", "7.1.0"),
+			ImportStateId:           fmt.Sprintf("%s/%s", namespace, name),
+			ResourceName:            "helm_release.imported",
+			ImportState:             true,
+			ImportStateVerify:       true,
+			ImportStateVerifyIgnore: []string{"set", "set.#", "repository"},
+			Check: resource.ComposeAggregateTestCheckFunc(
+				func() []resource.TestCheckFunc {
+					importTestCheckFuncs := []resource.TestCheckFunc{
+						resource.TestCheckResourceAttr("helm_release.imported", "metadata.0.revision", "1"),
+						resource.TestCheckResourceAttr("helm_release.imported", "metadata.0.version", "7.1.0"),
+						resource.TestCheckResourceAttr("helm_release.imported", "status", release.StatusDeployed.String()),
+						resource.TestCheckNoResourceAttr("helm_release.imported", "repository"),
+					}
+
+					// Test for default values being set
+					for key, value := range DefaultAttributes {
```

Comment:
Further to my comment above, I think using this variable means this acceptance test is relying upon an implementation detail of the provider. It would be better if we wrote out the checks explicitly. 

It's tedious, but it means at a glance you can look at the test and know exactly what is being tested.

Reply:
I agree with your assessment.